.changeset/little-frogs-fly.md

```diff
@@ -0,0 +1,5 @@
+---
+'xstate': patch
+---
+
+Fixed an issue with `EmittedFrom` sometimes not being able to infer the snapshot types from machines.
```

Comment:
Perhaps this sentence could be simplified/rephrased by a native speaker: @davidkpiano :!wink: :!wink: